src/language/typescript.ts

```diff
@@ -218,10 +220,29 @@ const serializePath = (url: string, item: TPathItemObject, rootName: string, cwd
 const serializeSchemaObject = (schema: TSchemaObject, rootName: string, cwd: string): TSerializedType => {
 	switch (schema.type) {
 		case undefined: {
-			const type = `${schema.$ref.replace(/^#\/definitions\//g, '')}`;
+			const $ref = schema.$ref;
+			const defBlock = fromNullable($ref.match(/#\/[^\/]+\//))
```

Comment:
What are you trying to achieve here? I believe everything can be rewritten with a single match

Reply:
replaced with a single match